Convert simple progress dialog to dialog fragment in GoogleSheetsUploaderActivity

grzesiek2010 opened the pull request:
Closes #3145 

#### What has been done to verify that this works as intended?
I wasn't able to reproduce the issue so I just tested the new implementation of the dialog in order to confirm that it works as before.

#### Why is this the best possible solution? Were any other approaches considered?
I refactored the progress dialog displayed during sending forms to Google Sheets, now it uses `DialogFragment` and dismissing it is safe.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
The progress dialog displayed during sending forms to Google Sheets has been refactored so it should be tested to confirm that it works with no regression. No difference should be visible and the pr is not related to anything else. 

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

grzesiek2010 committed:
Convert simple progress dialog to dialog fragment in GoogleSheetsUploaderActivity

mmarciniak90 commented:
@grzesiek2010 
I noticed an interesting case. I tried to send 200 submissions to GS.
When I send it on v1.22.3 or 1.22.4 everything finished with success. 
But then I did the same with these changes I got an error and part of submissions stayed on the device.
I checked this case on Android 7.0, 6.0 and 5.1.
This behavior requires more investigation so I will continue testing tomorrow.
![34304A9A-3F6F-46C2-9840-C363E47506E4](https://user-images.githubusercontent.com/3907554/59609998-c6ad4e80-9118-11e9-8659-ebade6abea3f.jpg)

grzesiek2010 commented:
That's weird...
There are some limits indeed: https://developers.google.com/sheets/api/limits 
> Google Sheets API has a limit of 500 requests per 100 seconds per project, and 100 requests per 100 seconds per user. Limits for reads and writes are tracked separately. 

but in this pr I just changed the way we build the progress bar I didn't touch the task which is responsible for sending forms.
The reason could be:
1. Faster sending forms, so you are reaching the limit of 100 requests per 100s
2. More requests that would cause you are reaching the limit as well

but as I said I didn't change anything, so it can't be a cause.

I updated the branch, could you try again and compare the results with the master branch?

mmarciniak90 commented:
@grzesiek2010 you have right. This strange behavior is visible on the master branch too, so I created a separate issue for it.

**Tested with success**

I did not notice regression 
Verified on Android 4.2, 4.4, 5.1, 6.0, 7.0, 8.1

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"
